tests/integration/targets/aws_sqs_queue/tasks/main.yml

```diff
@@ -0,0 +1,156 @@
+---
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+- environment:
+    AWS_ACCESS_KEY_ID: "{{ aws_access_key | default(omit) }}"
+    AWS_SECRET_ACCESS_KEY: "{{ aws_secret_key | default(omit) }}"
+    AWS_SESSION_TOKEN: "{{ security_token | default(omit) }}"
+    AWS_REGION: "{{ aws_region | default(omit) }}"
+
+  block:
+    - set_fact:
+        test_basedir: "{{ test_basedir | default(output_dir) }}"
+        resource_id: "{{ resource_prefix }}-sqs"
+    - set_fact:
+        test_role_name: "{{ resource_id }}-role"
+
+    - name: Copy terraform files to workspace
+      ansible.builtin.copy:
+        src: "{{ item }}"
+        dest: "{{ test_basedir }}/{{ item }}"
+      loop:
+        - cloud.tf
+
+    # iam_role_info - amazon.aws > 6.5.0 needed, tags need to be returned
+    - &verification
+      block:
+        - name: Fetch IAM role info
+          amazon.aws.iam_role_info:
+            name: "{{ test_role_name }}"
+          register: role_info
+
+        - name: Assert that there are {{ number_of_roles }} roles with role_name={{ test_role_name }}'
+          assert:
+            that:
+              - (role_info.iam_roles | length) == number_of_roles
+        # TODO: uncomment when iam_role_info does return tags
+        # - name: Assert that role {{ test_role_name }} is present and the info matches
+        #  assert:
+        #    that:
+        #      - role_info.iam_roles[0].tags.Name == "{{ resource_id }}"
+        #  when: number_of_roles == 1
```

Comment:
The amazon.aws 7.0.0 release is planned for Nov 2, but IMO we don't need this level of verification for this collection. The goal should be to test that terraform runs correctly and the module doesn't blow up. Definitely worth having a test for this bug, but you can just use the `cloud.terraform` module to do that.

Reply:
OK, removed dependency on amazon.aws 7.0.0. The cleanup task for role uses `amazon.aws.iam_role`, so I commented it out. It is not really needed, terraform with state=absent does cleanup.